scripts/charm_release.py

```diff
@@ -0,0 +1,229 @@
+        # Note(Reza): Currently SQA only supports the test for the amd64 architecture
+        # we should differentiate the TPIs for different architectures once arm64 is
+        # also supported. I have not put that in a file to avoid creating a perception 
+        # that more than one architecture could be tested. Having more than one arch
+        # would break the pipeline by creating duplicates as there are no ways to 
+        # distinguish test environments for architectures on SQA side. 
+        if arch != "amd64":
+            continue
+
+        track_state = TrackState()
+        for base in revision_matrix.get_bases():
+            revision = revision_matrix.get(arch, base)
+            if not revision:
+                continue
+
+            current_test_plan_instance_status = sqa.current_test_plan_instance_status(
+                charm_name, channel, revision
+            )
+            if not current_test_plan_instance_status:
+                if not dry_run:
+                    sqa.start_release_test(charm_name, channel, revision)
+                track_state.set_state(revision, sqa.TestPlanInstanceStatus.IN_PROGRESS)
+                continue
+
+            track_state.set_state(revision, current_test_plan_instance_status)
+
+    return track_state
+
+
+def process_track(track: str, dry_run: bool) -> ProcessState:
+    """Process the given track based on its current state."""
+
+    candidate_channel = f"{track}/candidate"
+    stable_channel = f"{track}/stable"
+
+    try:
+        candidate_revision_matrix = charmhub.get_revision_matrix(
+            "k8s", candidate_channel
+        )
+        print(f"Channel {candidate_channel} revisions:")
+        print(candidate_revision_matrix)
+
+        stable_revision_matrix = charmhub.get_revision_matrix("k8s", stable_channel)
+        print(f"Channel {stable_channel} reversions:")
+        print(stable_revision_matrix)
+    except HTTPError as e:
+        print(f"failed to get charm revisions: {e}")
+        return ProcessState.PROCESS_CI_FAILED
+
+    if not candidate_revision_matrix:
+        print(f"The channel {candidate_channel} has no revisions. Skipping...")
+        return ProcessState.PROCESS_UNCHANGED
+
+    if candidate_revision_matrix == stable_revision_matrix:
+        print(
+            f"The channel {candidate_channel} is already published in {stable_channel}. Skipping..."
+        )
+        return ProcessState.PROCESS_UNCHANGED
+
+    try:
+        state = ensure_track_state(
+            "k8s", candidate_channel, candidate_revision_matrix, dry_run
+        )
+        print(f"Track {track} is in state: {state}")
+
+        if state.succeeded:
+            print(f"Release run for {track} succeeded. Promoting charm revisions...")
+            if not dry_run:
+                charmhub.promote_charm("k8s", candidate_channel, stable_channel)
+                charmhub.promote_charm("k8s-worker", candidate_channel, stable_channel)
```

Comment:
I don't see you collecting the revision matrix at all for `k8s-worker`.  Why do its revisions get promoted along with `k8s`?